src/sinks/datadog/traces/request_builder.rs

```diff
@@ -391,3 +395,39 @@ impl DatadogTracesEncoder {
         }
     }
 }
+
+fn get_apm_stats_request(
+    key: &PartitionKey,
+    events: &[TraceEvent],
+    compression: Compression,
+    default_api_key: &Arc<str>,
+) -> Result<(RequestMetadata, Bytes), RequestBuilderError> {
+    let payload = stats::compute_apm_stats(key, events);
+    let encoded_payload =
+        rmp_serde::to_vec_named(&payload).map_err(|e| RequestBuilderError::FailedToEncode {
+            message: "APM stats encoding failed.",
+            reason: e.to_string(),
+            dropped_events: 0,
+        })?;
+    let uncompressed_size = encoded_payload.len();
+    let metadata = RequestMetadata {
+        api_key: key
+            .api_key
+            .clone()
+            .unwrap_or_else(|| Arc::clone(default_api_key)),
+        batch_size: 0,
```

Comment:
This is because this request does not really represents outgoing events but rather aggregated very specific metadata, but as it still generates traffic we want to account for the bytes sent.